// import test from 'ava';
test = require('ava')



async function prepare_all(){
    var createModule = require('./pyjs_runtime_node.js')
    var pyjs_module = await createModule()
    global.Module = pyjs_module
    await import('./python_data.js');
    var deps = await Module['_wait_run_dependencies']()
    return pyjs_module
}


var pyjs_promise = prepare_all()




test('test-loading-data',async function (t) {

    console.log("require pyjs_runtime_node.js")

    var pyjs = await pyjs_promise
    var interpreter =  new pyjs.Interpreter()
    var main_scope = pyjs.main_scope()

    try{
        interpreter.exec("raise RuntimeError('this error is raised and caught on purpose!')", main_scope)
    } catch(e){
        console.error(pyjs.get_last_exception())
    }

    t.is(1,1)

    main_scope.delete()
    interpreter.delete()

});
